pkg/services/provisioning/dashboards/file_reader.go

```diff
@@ -280,3 +291,41 @@ func (fr *fileReader) readDashboardFromFile(path string, lastModified time.Time,
 
 	return dash, nil
 }
+
+func newProvisioningSanityChecker() provisioningSanityChecker {
+	return provisioningSanityChecker{
+		uidUsage:   map[string]uint8{},
+		titleUsage: map[string]uint8{}}
+}
+
+type provisioningSanityChecker struct {
+	uidUsage   map[string]uint8
+	titleUsage map[string]uint8
+}
+
+func (checker provisioningSanityChecker) track(provisioningMetadata map[string]string) {
+	dashUid := provisioningMetadata["uid"]
+	dashTitle := provisioningMetadata["title"]
+	if len(dashUid) > 0 {
+		checker.uidUsage[dashUid] += 1
+	}
+	if len(dashTitle) > 0 {
+		checker.titleUsage[dashTitle] += 1
+	}
+
+}
+
+func (checker provisioningSanityChecker) logWarnings(log log.Logger) {
+	for uid, times := range checker.uidUsage {
+		if times > 1 {
+			log.Error("the same 'uid' is used more than once", "uid", uid)
+		}
+	}
+
+	for title, times := range checker.titleUsage {
+		if times > 1 {
+			log.Error("the same 'title' is used more than once", "title", title)
```

Comment:
is it per-folder?

Reply:
Good catch. I was incorrectly assuming folder to be specified in the provisioning config but that's not always the case. Thanks.

Reply:
It turns out that it actually isn't possible to set folderId in a provisioned dashboard, which makes sense as provisioning shouldn't depend on id's. It's still possible to have several provisioning configs using the same folder but solving that is out of scope of the current change.